mypy/subtypes.py

```diff
@@ -1620,6 +1620,18 @@ def are_parameters_compatible(
         return True
     trivial_suffix = is_trivial_suffix(right) and not is_proper_subtype
 
+    # def _(*a: Unpack[tuple[Any, ...]]) allows any number of arguments, not just infinite.
+    if right_star and isinstance(right_star.typ, UnpackType):
+        right_star_inner_type = get_proper_type(right_star.typ.type)
+        trivial_varargs = (
+            isinstance(right_star_inner_type, Instance)
+            and right_star_inner_type.type.fullname == "builtins.tuple"
+            and len(right_star_inner_type.args) == 1
+            and isinstance(get_proper_type(right_star_inner_type.args[0]), AnyType)
+        )
+    else:
+        trivial_varargs = False
+
     if (
         right.arg_kinds == [ARG_STAR]
         and isinstance(get_proper_type(right.arg_types[0]), AnyType)
```

Comment:
I'm pretty sure this check is redundant. It would require updating above to just handle `*args: Any` and handling `tuple[Any, ...]` here:

https://github.com/python/mypy/blob/e05770d899874abbe670caa2ddc888d7f41a6116/mypy/types.py#L2164-L2168

I'm just not sure because the comment says the form isn't constructed by mypy but it's the constructed form of any erased typevartuple? I also don't like treating any given `*args: Any` as finite length, but I think that ship has sailed.